src/oic/utils/authn/client.py

```diff
@@ -381,7 +381,7 @@ def verify(self, areq, **kwargs):
 
         try:
             # There might not be a client_id in the request
-            assert str(cid) in self.cli.cdb  # It's a client I know
+            assert str(cid) in self.cli.cdb  # nosec, It's a client I know
```

Comment:
I guess we should rewrite this to raise an `AuthnFailure` as well?

Reply:
yes. The cdb __contains__ method already checks for the KeyError, so that try is also bogus.